sdk/src/azure/core/az_json_writer.c

```diff
@@ -829,26 +825,18 @@ static AZ_NODISCARD az_result _az_json_writer_append_literal(
     remaining_json = az_span_copy_u8(remaining_json, ',');
   }
 
-  remaining_json = az_span_copy(remaining_json, literal);
+  az_span_copy(remaining_json, literal);
 
   _az_update_json_writer_state(ref_json_writer, required_size, required_size, true, literal_kind);
   return AZ_OK;
 }
 
 AZ_NODISCARD az_result az_json_writer_append_bool(az_json_writer* ref_json_writer, bool value)
 {
-  az_result result;
-  if (value)
-  {
-    result = _az_json_writer_append_literal(
-        ref_json_writer, AZ_SPAN_FROM_STR("true"), AZ_JSON_TOKEN_TRUE);
-  }
-  else
-  {
-    result = _az_json_writer_append_literal(
-        ref_json_writer, AZ_SPAN_FROM_STR("false"), AZ_JSON_TOKEN_FALSE);
-  }
-  return result;
+  return value ? _az_json_writer_append_literal(
+             ref_json_writer, AZ_SPAN_FROM_STR("true"), AZ_JSON_TOKEN_TRUE)
+               : _az_json_writer_append_literal(
+                   ref_json_writer, AZ_SPAN_FROM_STR("false"), AZ_JSON_TOKEN_FALSE);
```

Comment:
why is this better than before?
looks easier to read and follow like before.
Is there another way to fix this?  What about assigning an initial value to `result` ?  
Or calling return inside each condition?

Reply:
Yes, there are other ways to fix this (like you suggested). Given this method is fairly simple, I like the conciseness of the ternary operation here.

The issue with multiple returns is in non-optimized compilers, there is more code size bloat.